lib/internal/source_map/source_map_cache.js

```diff
@@ -175,7 +175,11 @@ function sourcesToAbsolute(baseURL, data) {
 // Move source map from garbage collected module to alternate key.
 function rekeySourceMap(cjsModuleInstance, newInstance) {
   const sourceMap = cjsSourceMapCache.get(cjsModuleInstance);
-  if (sourceMap) {
+  // If an exception occurs before a module finishes loading it will removed
+  // from the module cache and our WeakMap. To allow a source map to still be
+  // applied to the stack trace when this happens, we rekey on the thrown error.
+  // This is only possible if the error is an object and not a primitive type.
+  if (sourceMap && typeof newInstance === 'object') {
```

Comment:
Could `newInstance` be `null`, and if so, wouldn't that cause an error? Can you add a test case please?

Reply:
@aduh95 I did some testing it no longer seems like that rekey logic is needed to ensure that source maps are written to disk prior to exit. Perhaps this is because of refactors that have taken place related to source map serialization? Or perhaps a `FinalizationRegistry` fires on the next tick, such that the source map will be available while the process shuts down?

Either way, no tests fail if we simply remove the rekey logic.

Reply:
Nice! I still think it'd be interesting to have a test case with a `throw null` for completeness sake.